app/uk/gov/hmrc/economiccrimelevyregistration/services/KnownFactsQueuePullScheduler.scala

```diff
@@ -35,40 +38,53 @@ class KnownFactsQueuePullScheduler @Inject() (
   knownFactsQueueRepository: KnownFactsQueueRepository,
   enrolmentStoreProxyConnector: EnrolmentStoreProxyConnector
 )(implicit executionContext: ExecutionContext)
-    extends Logging {
+    extends Logging
+    with ErrorHandler {
 
   actorSystem.scheduler.scheduleAtFixedRate(initialDelay = 10.seconds, interval = 1.minute) { () =>
     processKnownFacts
   }
 
-  def processKnownFacts: Future[Unit] = {
+  def processKnownFacts: EitherT[Future, ResponseError, Unit] = {
     logger.info("Processing known facts for failed enrolments")
-
-    knownFactsQueueRepository
-      .pullOutstanding(
-        Instant.now(),
-        Instant.now()
-      )
-      .flatMap {
-        case None                     =>
-          logger.info("No known facts to process for failed enrolments")
-          Future.unit
-        case Some(knownFactsWorkItem) =>
-          enrolmentStoreProxyConnector
-            .upsertKnownFacts(
-              UpsertKnownFactsRequest(
-                verifiers = Seq(KeyValue(EclEnrolment.VerifierKey, knownFactsWorkItem.item.eclRegistrationDate))
-              ),
-              knownFactsWorkItem.item.eclReference
-            )(HeaderCarrier())
-            .map {
-              case Left(e)  =>
-                logger.error(s"Failed to upsert known facts for failed enrolment: ${e.message}")
-                knownFactsQueueRepository.markAs(knownFactsWorkItem.id, ProcessingStatus.Failed)
-              case Right(_) =>
-                logger.info("Successfully upserted known facts for failed enrolment")
-                knownFactsQueueRepository.completeAndDelete(knownFactsWorkItem.id)
-            }
-      }
+    for {
+      workItem <- pullOutstandingWorkItem.asResponseError
+      _        <- upsertKnownFacts(workItem).asResponseError
+    } yield ()
   }
+  private def pullOutstandingWorkItem: EitherT[Future, KnownFactsError, WorkItem[KnownFactsWorkItem]] =
+    EitherT {
+      knownFactsQueueRepository
+        .pullOutstanding(Instant.now(), Instant.now())
```

Comment:
not sure what pullOutstanding is doing but you may want to just create a value for now and pass that in
```
val now = Instant.now()
knownFactsQueueRepository
         .pullOutstanding(now, now)
```

Reply:
Done :)